tests/sceneManager_test.cpp

```diff
@@ -11,6 +11,68 @@ using objects::Object;
 using objects::TriangleObj;
 
 const double kSkipFreq = 1000;
+const core::Vec3 kSkipVec3(0, 0, 0);
 generators::FakeRandomGen fakeRand;
 
-// TODO: Change all doubles to float
+class SceneManagerTest : public ::testing::Test
+{
+protected:
+    SceneManager manager;
+    std::vector<objects::EnergyCollector *> energyCollectors;
+    double energyCollectorRadius = objects::EnergyCollector(kSkipVec3).getRadius();
+    void SetUp() override
+    {
+        energyCollectors = manager.getEnergyCollectors();
+    }
+
+    bool performHitAtEnergyCollectors(const Ray &tempRay, RayHitData *hitData)
+    {
+        bool hit = false;
+        for (const auto &energyColPtr : energyCollectors)
```

Comment:
Do not use auto where it does not improve the readability of the code. If you actually spelled out the type, you would not need Ptr at the end.

In this case these are pointers so you can just copy them.

Reply:
done